polyfill/spatial-navigation-polyfill.js

```diff
@@ -269,8 +269,13 @@
       const container = bestCandidate.getSpatialNavigationContainer();
 
       // Scrolling container or document when the next focusing element isn't entirely visible
-      if (isScrollContainer(container) && !isEntirelyVisible(bestCandidate))
-        bestCandidate.scrollIntoView();
+      if (isScrollContainer(container) && !isEntirelyVisible(bestCandidate)) {
+        createSpatNavEvents('beforescroll', bestCandidate, null, dir);
+
+        if (!navbeforescrollPrevented) {
```

Comment:
This condition needs to be changed:
If the `scroll` event is prevented, then `scrollIntoView` won't be triggered.